layers/+lang/emacs-lisp/packages.el

```diff
@@ -46,6 +47,24 @@
 (defun emacs-lisp/post-init-eldoc ()
   (add-hook 'emacs-lisp-mode-hook 'eldoc-mode))
 
+(defun emacs-lisp/init-auto-compile ()
+  (use-package auto-compile
+    :defer t
+    :init
+    (progn
+      (setq auto-compile-display-buffer nil
+            auto-compile-use-mode-line t
+            auto-compile-mode-line-counter t)
+      (add-hook 'emacs-lisp-mode-hook
+                (lambda () (auto-compile-on-save-mode 1)))
+      (add-hook 'lisp-interaction-mode-hook
+                (lambda () (auto-compile-on-save-mode 1))))
```

Comment:
Agreed on the `lisp-interaction-mode`. The docstring for `auto-compile-mode` says to use the global version. If I just use `(auto-compile-on-save-mode 1)` then the code won't be deferred, so I was using this as a cheap trick to allow the package to only load when an elisp buffer is visited. It probably doesn't make a difference though so I can change it.